hail/Makefile

```diff
@@ -21,23 +19,14 @@ HAIL_PYTHON3 ?= python3
 PIP ?= $(HAIL_PYTHON3) -m pip
 ifdef JAVA_HOME
 JAVAC ?= $(JAVA_HOME)/bin/javac
-JAR ?= $(JAVA_HOME)/bin/jar
 else
 JAVAC ?= javac
-JAR ?= jar
 endif
 
 PYTEST_TARGET ?= test/hail test/hailtop
 DOCTEST_TARGET ?= hail
 PYTEST_INTER_CLOUD_TARGET ?= test/hailtop/inter_cloud
 
-# not perfect, not robust to simdpp changes, but probably fine
-BUILD_DEBUG_PREFIX := build/classes/scala/debug
-JAR_SOURCES := $(shell git ls-files src/main)
-JAR_SOURCES += build.gradle
-JAR_TEST_SOURCES := $(shell git ls-files src/test)
-JAR_DEBUG_SOURCES := src/debug/scala/is/hail/annotations/Memory.java
-JAR_DEBUG_CLASSES := $(addprefix $(BUILD_DEBUG_PREFIX)/, $(JAR_DEBUG_SOURCES:src/debug/scala/%.java=%.class))
```

Comment:
We should keep the source/test dependencies on the jar targets, otherwise we may not rerun them when files change.